### ISSUE TITLE
config_scene: app stays locked to the main scene even when a controller is unplugged

### ISSUE THREAD

**Issue by calexil** [calexil/FightstickDisplay#74]
introduced by this commit:

https://github.com/calexil/FightstickDisplay/commit/6e211b63b2e59004e1f91fe79ab400a53bca68af

**Comment by calexil** [comment]
this is happening now because the app is no longer checking constantly for a controller as it did previously

**Comment by benmoran56** [comment]
This is a tricky one. The last way we did it was obviously a big hack.

If we want to do it right, we need to actually monitor for device changes (not just keep checking if controllers exist). 
On Linux, this just just a matter of watching for changes to `/proc/bus/input/devices`, and then re-scanning the Controllers only when it does. How does that sound? pyglet doesn't _yet_ have this functionality built-in, but it something that will appear in the future. When that happens, we could remove the custom code from FightstickDisplay. How does that sound?

**Comment by calexil** [comment]
makes sense

**Comment by calexil** [comment]
@benmoran56  Hey, just wanted to let you know I managed to take back the pyglet subreddit here: https://www.reddit.com/r/pyglet/

I have already got the css setup(I think it looks nice at least), rules in place, etc. But wanted to get you on the moderation team if you like, so send me your reddit username and any other team members and I'll add them as well

**Comment by benmoran56** [comment]
Sorry for leaving this hanging.  pyglet now has built-in support for hot-plugging Controllers, so I updated fightstick.py to take advantage of it.  I've tested it on my own machine with two Controllers, and it seems to work well. The deadzone settings are also working well. 

On Linux, pyglet now does auto-detection for Controller layouts, in accordance to the Linux GamePad specification:
https://www.kernel.org/doc/html/v4.13/input/gamepad.html#detection
This means that FightstickDisplay will now work for any Controller, even if it doesn't have a specific mapping available.
Linux drivers for new devices should all adhere to this specification, so the layouts should be consistent. Legacy drivers might have funky layouts, but at least they should work. 

On Reddit, 
I actually stopped using Reddit a while ago, but that's good news. I'll pick some folks on the Discord to see if anyone wants to help out.
